frontend/src/components/forms/SvgButton.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import classNames from 'classnames';
+import EditIcon from '@material-ui/icons/EditOutlined';
+import SettingsSharpIcon from '@material-ui/icons/SettingsSharp';
+import ForumIcon from '@material-ui/icons/ForumOutlined';
+import InfoIcon from '@material-ui/icons/InfoOutlined';
+import DeleteOutlineOutlinedIcon from '@material-ui/icons/DeleteOutlineOutlined';
+import DeleteIcon from '@material-ui/icons/DeleteSharp';
+import CloseIcon from '@material-ui/icons/Close';
+
+import colors from '../../colors.module.scss';
+import css from './SvgButton.module.scss';
+
+const classes = classNames.bind(css);
+
+type ButtonProps = {
+  onClick?: (event: React.MouseEvent) => void;
+};
+
+type ExtraProps = {
+  hoverColor?: string;
+  className?: string;
+};
+
+type SvgButton<T = {}> = React.FC<T & ButtonProps & { href?: string }>;
+
+function svgButton<E = {}>(
+  Icon: React.FC<ButtonProps>,
+  {
+    hoverColor = colors.black60,
+    className,
+    ...extraProps
+  }: ExtraProps & { [K in keyof E]?: E[K] },
+): SvgButton<Pick<E, Exclude<keyof E, keyof ExtraProps>>> {
```

Comment:
Which props get picked here is hard to read for me (is only hoverColor overridden by ExtraProps in Pick?), so perhaps setting the default color in scss could make this a bit simpler?
`fill: var(--icon-hover-color, $COLOR_BLACK60);`
https://developer.mozilla.org/en-US/docs/Web/CSS/var()#custom_properties_with_fallbacks_for_use_when_the_property_has_not_been_set

Reply:
I moved the default color to css, and renamed the `extraProps` parameter to a better describing `defaults`, as it can provide defaults for any optional properties in `E`.

The `Pick` will exclude any properties in `ExtraProps` from the resulting button's props, e.i. `hoverColor` and `className`.